docs/source/caching/cache-key-resolution.mdx

```diff
@@ -175,4 +175,22 @@ public enum SchemaConfiguration: ApolloAPI.SchemaConfiguration {
     return nil
   }
 }
-```
\ No newline at end of file
+```
+
+## Caveats & limitations
+
+Because cache key resolution is performed both on raw JSON (from a network response or cache hit) and `SelectionSet` model data (when writing to the cache directly), the underlying `object` will have different formats.
+
+This means that cache key resolution has a few notable limitations:
+
+1. Computing cache keys from fields on nested objects is only allowed if the nested object
+does not have its own cache key. If the nested object has its own cache key, it will be
+normalized as a seperate cache entity. Cache keys for entities cannot be dependent on
+other entities.
+
+2. When computing a cache key using a field of a `CustomScalarType`, the underlying type of
+the value in the `object` dictionary will vary. It may be the raw JSON value for the scalar
+(when the source is a network response or cache hit) or the specific custom scalar type for
+the field (when the source is a `SelectionSet` model to write to the cache). When using a
+custom scalar field to compute a cache key, make sure to check the type and handle both of
+these possibilities
```

Comment:
This is a good start to the docs. I think it would be helpful to add examples here of how to work with the cache data from the various sources and how to handle the different formats.

Reply:
Definitely agree! This is just the bare minimum for Beta. I'm going to add more details and usage examples for this as well as for the new initializers next week.